backend/timed/projects/views.py

```diff
@@ -177,6 +178,8 @@ class TaskViewSet(ModelViewSet):
         (
             # superuser may edit all tasks
             IsSuperUser
+            # only superuser can update tasks of archived projects
+            | (IsUpdateOnly & IsProjectActive)
             # managers may edit all tasks
             | IsManager
```

Comment:
The order is wrong!
This means that anyone who isn't a superuser can update a task if the project is active.

Reply:
Right - and wrong. It's not the order, but the full expression for the permission

```suggestion
            # only superuser can update tasks of archived projects
            | (IsUpdateOnly & IsProjectActive & IsManager)
```